.travis.yml

```diff
@@ -0,0 +1,124 @@
+language: cpp
+
+# sudo: false instructs travis to build our project in a docker VM (faster)
+# Can not yet install fglrx packages with 'false'
+sudo: required # false
+
+# os: expands the build matrix to include multiple os's
+os:
+  - linux
+  - osx
+
+# compiler: expands the build matrix to include multiple compilers (per os)
+compiler:
+  - gcc
+  - clang
+
+addons:
+  # apt: is disabled on osx builds
+  # apt: needed by docker framework to install project dependencies without
+  # sudo.  Apt uses published Ubunto PPA's from https://launchpad.net/
+  # https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json
+  apt:
+    sources:
+      # ubuntu-toolchain-r-test contains newer versions of gcc to install
+      - ubuntu-toolchain-r-test
+      # llvm-toolchain-precise-3.6 contains newer versions of clang to install
+      # - llvm-toolchain-precise-3.6
+      # kubuntu-backports contains newer versions of cmake to install
+      - kubuntu-backports
+      # boost-latest contains boost v1.55
+      # - boost-latest
+    packages:
+      # g++-4.8 is minimum version considered to be the first good c++11 gnu compiler
+      - g++-4.8
+      # - clang-3.6
+      # We require v2.8.12 minimum
+      - cmake
+      # I'm finding problems between pre-compiled versions of boost ublas, with gtest
+      # stl_algobase.h: error: no matching function for call to swap()
+      # - libboost-program-options1.55-dev
+      # - libboost-serialization1.55-dev
+      # - libboost-filesystem1.55-dev
+      # - libboost-system1.55-dev
+      # - libboost-regex1.55-dev
+      # The package opencl-headers on 'precise' only installs v1.1 cl headers; uncomment for 'trusty' or greater
+#      - opencl-headers
+      # Uncomment one of the following when fglrx modules are added to the apt whitelist
+#      - fglrx
+#      - fglrx=2:8.960-0ubuntu1
+#      - fglrx=2:13.350.1-0ubuntu0.0.1
+
+# env: specifies additional global variables to define per row in build matrix
+env:
+  global:
+    - CLSPARSE_ROOT=${TRAVIS_BUILD_DIR}/bin/make/release
+
+# The following filters our build matrix; we are interested in linux-gcc & osx-clang
+matrix:
+  exclude:
+    - os: linux
+      compiler: clang
+    - os: osx
+      compiler: gcc
+
+before_install:
+  # Remove the following linux clause when fglrx can be installed with sudo: false
+  - if [ ${TRAVIS_OS_NAME} == "linux" ]; then
+      sudo apt-get update -qq &&
+      sudo apt-get install -qq fglrx=2:13.350.1-0ubuntu0.0.1;
+    fi
+  - if [ ${TRAVIS_OS_NAME} == "linux" ]; then
+      export OPENCL_ROOT="${TRAVIS_BUILD_DIR}/opencl-headers";
+      export BUILD_BOOST="ON";
+    fi
+  - if [ ${TRAVIS_OS_NAME} == "osx" ]; then
+      brew update;
+      brew outdated boost || brew upgrade boost;
+      brew outdated cmake || brew upgrade cmake;
+      export BUILD_BOOST="OFF";
+    fi
+  - if [ ${CXX} = "g++" ]; then export CXX="g++-4.8" CC="gcc-4.8"; fi
```

Comment:
It seems to me that this is fixable bug. Im running Linux Mint (Ubuntu
based distro) in my system ${CC} variable is points to /usr/bin/cc. I've
run standalone boost bootstrap.sh with flag --with-toolset=cc and it
failed. If you change this flag to --with-toolset=gcc it will compile
correctly. Other option is to run bootstrap without --with-toolset flag
then at my linux configuration boost automatically discovers gcc compiler.
When I took a peek at previous ExternalBoost.cmake it seems that we did not
define this flag at all.

I don't know what documentation says. Could you please send me a link to
that?


2015-08-14 16:01 GMT+02:00 Kent Knox <notifications@github.com>:

> In .travis.yml
> <https://github.com/clMathLibraries/clSPARSE/pull/122#discussion_r37078691>
> :
>
> > +  # Remove the following linux clause when fglrx can be installed with sudo: false
> > +  - if [ ${TRAVIS_OS_NAME} == "linux" ]; then
> > +      sudo apt-get update -qq &&
> > +      sudo apt-get install -qq fglrx=2:13.350.1-0ubuntu0.0.1;
> > +    fi
> > +  - if [ ${TRAVIS_OS_NAME} == "linux" ]; then
> > +      export OPENCL_ROOT="${TRAVIS_BUILD_DIR}/opencl-headers";
> > +      export BUILD_BOOST="ON";
> > +    fi
> > +  - if [ ${TRAVIS_OS_NAME} == "osx" ]; then
> > +      brew update;
> > +      brew outdated boost || brew upgrade boost;
> > +      brew outdated cmake || brew upgrade cmake;
> > +      export BUILD_BOOST="OFF";
> > +    fi
> > +  - if [ ${CXX} = "g++" ]; then export CXX="g++-4.8" CC="gcc-4.8"; fi
>
> @jpola <https://github.com/jpola>
> Thank you for running tests; this build works on centos6.6, but I assume
> that my ${CC} variable is not 'cc'. Do you think we have a fixable bug? Or
> is this a documentation issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/clMathLibraries/clSPARSE/pull/122/files#r37078691>.
>